proxy/proxy.ts

```diff
@@ -0,0 +1,119 @@
+// @ts-ignore
+import isMartianIP from 'martian-cidr'
+import type http from 'node:http'
+import { createServer } from 'node:http'
+import { styleText } from 'node:util'
+
+/**
+ * Creates proxy server
+ *
+ * isProduction - main toggle for production features:
+ * - will allow only request that match productionDomainSuffix
+ *
+ * silentMode - will silent the output. Useful for testing
+ *
+ * hostnameWhitelist - Sometimes you need to allow request to certain ips like localhost for testing purposes.
+ *
+ * productionDomainSuffix - if isProduction, then only request from origins that match this param are allowed
+ *
+ * @example
+ * const p = createProxyServer()
+ * p.listen(5555).then(() => console.log('running on 5555'))
+ *
+ * @param { isProduction, silentMode, hostnameWhitelist, productionDomainSuffix } config
+ */
+const createProxyServer = (
+  config: {
+    hostnameWhitelist?: string[]
+    isProduction?: boolean
+    productionDomainSuffix?: string
+    silentMode?: boolean
+  } = {}
+): http.Server => {
+  let isProduction = config.isProduction || false
+  let silentMode = config.silentMode || false
+  let hostnameWhitelist = config.hostnameWhitelist || []
+  let productionDomainSuffix = config.productionDomainSuffix || ''
+
+  return createServer(async (req, res) => {
+    let url = decodeURIComponent(req.url!.slice(1))
+    let sent = false
+
+    try {
+      // Only http or https protocols are allowed
+      if (!url.startsWith('http://') && !url.startsWith('https://')) {
+        throw new Error('Only http or https requests are allowed.')
```

Comment:
It will be nice to have `400` (Bad request) response here. It seems for me that `Error` is always will return `500`,

Reply:
Agreed, added 400 response here

Reply:
Agreed, just fixed that